reconcile/ocm_machine_pools.py

```diff
@@ -262,7 +262,9 @@ def has_diff(self, pool: ClusterMachinePoolV1) -> bool:
             or self.taints != pool.taints
             or self.labels != pool.labels
             or self.aws_node_pool.instance_type != pool.instance_type
-            or self.subnet != pool.subnet
+            # if the nodepool in app-interface does not define a subnet explicitely
+            # we don't consider it a diff. we don't manage it
+            or (pool.subnet is not None and self.subnet != pool.subnet)
```

Comment:
so this means we'll have non matching data between app-interface and the reality ?

Reply:
it was not really non-matching data. not specifying them in app-interface means "i don't manage this".

i added the promised phase 2 to this MR in the meantime to open an app-interface MR to patch the missing subnets.

Reply:
we can and we should specify subnets for day2 node pool operations. but at installtime we can't influence it and we end up with a set of pools where ROSA decides how to spread them accross the available private subnets.

we can improve on this behaviour once there is the ability to create HCP clusters without node pools at install time. becaues then we wait until the cluster is ready and `ocm-machine-pools` can create the pools in the explicitely defined subnets.

Reply:
> > Does that mean we should never apply subnet from app-interface to ocm? When app-interface pool subnet is not None and we have a different subnet value in current pool, this condition will be true, and show as diff, is this expected?
> 
> once a node pool has been created, its subnet ID can't be changed anymore. at this point OCM becomes the source of truth for it. we implemented the notion of an `invalid_diff` to highlight the problem, so it is expected that `has_diff` shows a diff in subnet between a-i and OCM.

So for the use case of `nodepools have been recreated on OCM side without app-interface involvement`, this diff will show in `has_diff` but will fail `invalid_diff` check, the integration will fail until patch mr merged, is this expected?

Reply:
yes, it still makes us aware and we need to intervene. we should not do things manually in OCM.